concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs

```diff
@@ -352,6 +352,14 @@ deriving via (LiftMonadBlobStore (StateT s) m)
 deriving via (LiftMonadBlobStore (ExceptT e) m)
     instance MonadBlobStore m => MonadBlobStore (ExceptT e m)
 
+-- |This instance lifts a 'MonadBlobStore' over a 'ReaderT' transformer.
+-- It is overlapped by the instance for @ReaderT BlobStore IO@ that uses
+-- the 'BlobStore' to provide the instance instead. This overlap should
+-- not be problematic, since there should be no 'MonadBlobStore' instance
+-- for 'IO'.
+deriving via (LiftMonadBlobStore (ReaderT r) m)
+    instance {-# OVERLAPPABLE #-} MonadBlobStore m => MonadBlobStore (ReaderT r m)
```

Comment:
Why is this actually needed? For the caching of instances? Why could it not have just stayed the way it was?

I think this is a bad instance due to the overlapping. It requires yet more understanding of yet more obscure Haskell stuff.

Reply:
Yes, it is for caching of Instances. This instance allows us to avoid a complex lifting/unlifting of the `ReaderT` monad (as by `cacheBufferedBlobbed`.

I agree that overlapping instances are not ideal, but I think removing the instance for `ReaderT BlobStore IO` is a better solution.